### PR TITLE
Add  TC OCP-65063 Create single-az machinepool for multi-az cluster

### PR DESCRIPTION
**Pull request by amalykhi**
https://polarion.engineering.redhat.com/polarion/#/project/OSE/workitem?id=OCP-65063

terraform-provider-rhcs % ginkgo run --label-filter feature-machinepool  ./tests/e2e
......
an 6 of 17 Specs in 60.612 seconds
SUCCESS! -- 6 Passed | 0 Failed | 0 Pending | 11 Skipped
PASS

### CODE REVIEW

**Comment by amalykhi** [comment]
/pj-rehearse periodic-ci-terraform-redhat-terraform-provider-rhcs-main-rosa-sts-ad-critical-high-f3

File: tests/e2e/machine_pool_test.go [hunk 1]
```diff
@@ -19,8 +19,8 @@ var _ = Describe("TF Test", func() {
 			mpService = exe.NewMachinePoolService(con.MachinePoolDir)
 		})
 		AfterEach(func() {
-			err := mpService.Destroy()
-			Expect(err).ToNot(HaveOccurred())
+			//err := mpService.Destroy()
```
Comment (xueli181114) [inline tests/e2e/machine_pool_test.go:22]: Why destroy commented?

Comment (amalykhi) [reply tests/e2e/machine_pool_test.go]: It is left from debugging. Fixed

**Comment by amalykhi** [comment]
@xueli181114 I fixed the commented line. Please review again

File: tests/e2e/machine_pool_test.go [hunk 2]
```diff
@@ -353,6 +353,78 @@ var _ = Describe("TF Test", func() {
...
+		Context("Author:amalykhi-High-OCP-65063 @OCP-65063 @amalykhi", func() {
+			It("Author:amalykhi-High-OCP-65063 Create single-az machinepool for multi-az cluster", ci.Day2, ci.High, ci.FeatureMachinepool, func() {
+				if !profile.MultiAZ {
+					Skip("The test is configured for MultiAZ cluster only")
+				}
+				getResp, err := cms.RetrieveClusterDetail(ci.RHCSConnection, clusterID)
+				Expect(err).ToNot(HaveOccurred())
+				azs := getResp.Body().Nodes().AvailabilityZones()
+				By("Create additional machinepool with availability zone specified")
+				replicas := 3
```
Comment (xueli181114) [inline tests/e2e/machine_pool_test.go:365]: Better to use replicas 1 or 2 for the testing which will cover the replicas validation well.

File: tests/e2e/machine_pool_test.go [hunk 3]
```diff
@@ -353,6 +353,78 @@ var _ = Describe("TF Test", func() {
...
+				machineType := "r5.xlarge"
```
Comment (xueli181114) [inline tests/e2e/machine_pool_test.go:366]: Better to list the machineTypes from AWS and prepare it from the list result. There is aws_inquiries endpoint in OCM.

File: tests/e2e/machine_pool_test.go (hunk 2 above)
Comment (amalykhi) [reply tests/e2e/machine_pool_test.go]: fixed

File: tests/e2e/machine_pool_test.go (hunk 3 above)
Comment (amalykhi) [reply tests/e2e/machine_pool_test.go:366]: @xueli181114 I got it. May I refactor all the machinepool tests with this enhancement in the next PR?

Comment (xueli181114) [reply tests/e2e/machine_pool_test.go:366]: @amalykhi Of course. Then I will LGTM it.

**Pull request merged**